lib/vega_lite/data.ex

```diff
@@ -327,6 +327,311 @@ defmodule VegaLite.Data do
+    * `:grid_color` - backround color for the grid. Defaults to `"white"`.
+    * `:grid_stroke_color` - stroke color for the grid markings. Defaults to `"black"`.
+    * `:grid_stroke_opacity` - opacity for the grid strokes. Defaults to `1`.
+    * `:scheme` - the color scheme for multi-line plots. Defaults to `"turbo"`
+
+  ## Examples
+
+  In the example below, we plot a dataset by specifying stylized point marks and connecting it with
+  an interpolated line.
+
+      data = %{"r" => [1, 2, 3, 3, 4], "theta" => [0, 30, 45, 135, 270]}
+
+      VegaLite.Data.polar_plot(
+        VegaLite.new(title: "Polar Plot", height: 500, width: 500),
+        [
+          {data, :point, [stroke: "white", stroke_width: 3, tooltip: [data: true]]},
+          {data, :line, point: true, interpolate: "cardinal", color: "black", tooltip: [data: true]}
+        ],
+        angle_marks: [0, 15, 30, 45, 60, 90, 180, 270, 360],
+        radius_marks: [0, 1, 2, 3, 4, 5]
+      )
+
+  In the next example, we add a second line to the plot grouping by the `:legend_key`.
+  Note that the `:legend_key` will also be the legend title, so a readable name is desirable.
+
+      legend_key = "Line Groups"
+
+      data = %{
+        "r" => [1, 2, 3, 3, 4],
+        "theta" => [0, 30, 45, 135, 270],
+        legend_key => List.duplicate("First Line", 5)
+      }
+
+      other_data = %{
+        "r" => [8, 6, 4, 2, 0],
+        "theta" => [0, 30, 60, 90, 120],
+        legend_key => List.duplicate("Second Line", 5)
+      }
+
+      VegaLite.Data.polar_plot(
+        VegaLite.new(title: "Polar Plot", height: 500, width: 500),
+        [
+          {data, :point, [stroke: "black", stroke_width: 3]},
+          {data, :line, point: true, interpolate: "cardinal"},
+          {other_data, :point, [stroke: "black", stroke_width: 3]},
+          {other_data, :line, point: true, interpolate: "cardinal"}
+        ],
+        angle_marks: [0, 15, 30, 45, 60, 90, 180, 270, 360],
+        legend_key: legend_key
+      )
+  """
+  def polar_grid(vl \\ Vl.new(), opts \\ []) do
+    opts =
+      Keyword.validate!(
+        opts,
+        [
+          :radius_marks,
+          angle_marks: [0, 90, 180, 270, 360],
+          direction: :counter_clockwise,
+          angle_offset: 0,
+          opacity: 1,
+          color: "white",
+          stroke_color: "black",
+          stroke_opacity: 1
+        ]
+      )
+
+    unless opts[:radius_marks] do
+      raise ArgumentError, "missing :radius_marks option"
+    end
+
+    grid_layers = polar_angle_layers(opts)
+    radius_layers = polar_radius_layers(opts)
+
+    vl = append_layers(vl |> Vl.data_from_values(%{"_r" => [0]}), grid_layers ++ radius_layers)
+
+    put_in(
+      vl.spec["_vl_polar_config"],
+      opts
+      |> Keyword.take([:radius_marks, :angle_offset, :direction])
+      |> Map.new()
+    )
+  end
+
+  def polar_plot(vl \\ Vl.new(), data, mark, fields, opts \\ []) do
```

Comment:
We should not have fields and opts. Only fields is enough. We also need to only extract the desired fields from `data`. Please use `chart` as a starting point here. :)

Reply:
Done